packages/prettier-plugin-java/src/printers/lexical-structure.ts

```diff
@@ -7,10 +8,26 @@ import {
   IToken,
   LiteralCtx
 } from "java-parser";
+import { builders } from "prettier/doc";
+
+const { hardline } = builders;
 
 export class LexicalStructurePrettierVisitor extends BaseCstPrettierPrinter {
   literal(ctx: LiteralCtx) {
-    if (ctx.CharLiteral || ctx.TextBlock || ctx.StringLiteral || ctx.Null) {
+    if (ctx.TextBlock) {
+      const lines = ctx.TextBlock[0].image.split("\n");
+      const open = lines.shift()!;
+      const close = lines.pop()?.trim()!;
+      const baseIndent = Math.min(
```

Comment:
I think that the last line of the textBlock should be included in the baseIndent search:

Indeed, the two text blocks in the following snippet are not equals, if I'm not mistaken

```java
@Test
    void isFalse() {
        String s = """
        test
         a""";

        String s2 = """
        test
        a""";
        
        assertThat(s).isEqualTo(s2);
    }
```

Reply:
Thanks for catching this! You are correct, and I've pushed a fix.